build/source/engine/read_icond.f90

```diff
@@ -365,53 +296,67 @@ USE get_ixname_module,only:get_ixmvar,get_ixindex ! access function to find inde
  ! close file
  close(unt)
  ! set iLayerHeight for the bottom layer
- mvar_data%var(iLookMVAR%iLayerHeight)%dat(nLayers) = &
- mvar_data%var(iLookMVAR%iLayerHeight)%dat(nLayers-1) + mvar_data%var(iLookMVAR%mLayerDepth)%dat(nLayers)
+ prog_data%var(iLookPROG%iLayerHeight)%dat(nLayers) = &
+ prog_data%var(iLookPROG%iLayerHeight)%dat(nLayers-1) + prog_data%var(iLookPROG%mLayerDepth)%dat(nLayers)
  ! check matric head is read correctly
- !print*,'mLayerMatricHead ', mvar_data%var(iLookMVAR%mLayerMatricHead)%dat(:)
+ !print*,'mLayerMatricHead ', prog_data%var(iLookPROG%mLayerMatricHead)%dat(:)
  ! ***************************************************************************************
  ! ***************************************************************************************
  ! ensure the snow albedo is realistic
  ! ***************************************************************************************
  ! ***************************************************************************************
  ! ensure the spectral average albedo is realistic
- if(mvar_data%var(iLookMVAR%scalarSnowAlbedo)%dat(1) > mpar_data%var(iLookPARAM%albedoMax)) &
-    mvar_data%var(iLookMVAR%scalarSnowAlbedo)%dat(1) = mpar_data%var(iLookPARAM%albedoMax)
- if(mvar_data%var(iLookMVAR%scalarSnowAlbedo)%dat(1) < mpar_data%var(iLookPARAM%albedoMinWinter)) &
-    mvar_data%var(iLookMVAR%scalarSnowAlbedo)%dat(1) = mpar_data%var(iLookPARAM%albedoMinWinter)
+ if(prog_data%var(iLookPROG%scalarSnowAlbedo)%dat(1) > parData(iLookPARAM%albedoMax)) &
+    prog_data%var(iLookPROG%scalarSnowAlbedo)%dat(1) = parData(iLookPARAM%albedoMax)
```

Comment:
I've opened up a new issue to address this, and more generally to improve handing of initial conditions.